File: lib/core/models/cake.dart
class Cake {
  final String id;
  final String name;
  final String? description;
  final double price;
  final double discount;
  final String? imageUrl;

  Cake({required this.id, required this.name, this.description, required this.price, this.discount = 0, this.imageUrl});

  factory Cake.fromJson(Map<String, dynamic> json) {
    return Cake(
      id: json['_id'] as String? ?? json['id'] as String,
      name: json['name'] as String,
      description: json['description'] as String?,
      price: (json['price'] as num).toDouble(),
      discount: (json['discount'] as num? ?? 0).toDouble(),
      imageUrl: json['imageUrl'] as String?,
    );
  }
}

File: server/src/models/Cake.js
import mongoose from 'mongoose';

const cakeSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    description: { type: String, default: '' },
    ingredients: [{ type: String }],
    price: { type: Number, required: true },
    discount: { type: Number, default: 0 },
    categories: [{ type: String, index: true }],
    imageUrl: { type: String },
    publicId: { type: String },
    flavor: { type: String, index: true },
    type: { type: String, index: true },
    popularity: { type: Number, default: 0, index: true },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

export const Cake = mongoose.models.Cake || mongoose.model('Cake', cakeSchema);


